test/basics.jl

```diff
@@ -0,0 +1,45 @@
+string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UniStr]
+
+@testset "constructors" begin
+    for strtype in string_types
+        @test convert(strtype, [0x61,0x62,0x63,0x21]) == "abc!"
+        @test convert(strtype, "abc!") == "abc!"
+
+        @test isempty(string())
+        @test eltype(GenericString) == Char
```

Comment:
This doesn't test anything about the new strings, so it should be moved out of the loop (put in a commented out section for now)

Reply:
Just the `eltype` line?

Reply:
I'll leave the commented line in the loop for now.
I think it makes it easier to remember where it came from.
If we're sure we don't need it, we should remove it completely.